packages/core-components/src/components/CopyTextButton/CopyTextButton.test.tsx

```diff
@@ -21,6 +21,10 @@ import { renderInTestApp } from '@backstage/test-utils';
 import { CopyTextButton } from './CopyTextButton';
 import { ApiRegistry, ApiProvider } from '@backstage/core-app-api';
 import { errorApiRef, ErrorApi } from '@backstage/core-plugin-api';
+// should we add this to dev dependencies?
+// (already a dependency of react-use)
+// eslint-disable-next-line import/no-extraneous-dependencies
+import copy from 'copy-to-clipboard';
```

Comment:
I think this is a bit deep tbh, better just test that the function returned by the hook is being called

Reply:
Yeah it is. The idea was to test the entire flow more at "e2e" level, without having any intermediate mocking that might require splitting the test into multiple ones.

I've updated with your suggestions :)